cloudinit/sources/DataSourceLXD.py

```diff
@@ -223,14 +224,129 @@ def is_platform_viable() -> bool:
     return False
 
 
+def _do_request(
+    session: requests.Session, url: str, do_raise: bool = True
+) -> requests.Response:
+    response = session.get(url)
+    LOG.debug("[GET] [HTTP:%d] %s", response.status_code, url)
+    if do_raise and not response.ok:
+        raise sources.InvalidMetaDataException(
+            "Invalid HTTP response [{code}] from {route}: {resp}".format(
+                code=response.status_code,
+                route=url,
+                resp=response.text,
+            )
+        )
+    return response
+
+
+class _MetaDataReader:
+    def __init__(self, api_version: str = LXD_SOCKET_API_VERSION):
+        self.api_version = api_version
+        self._version_url = url_helper.combine_url(LXD_URL, self.api_version)
+
+    def _get_meta_data(self, session: requests.Session) -> dict:
```

Comment:
_get_meta_data doesn't really have a lot to it. Can we just avoid this method definition and call it inline.

Reply:
Done.